virtual_ecosystem/models/animals/animal_communities.py

```diff
@@ -131,14 +132,25 @@ def migrate(self, migrant: AnimalCohort, destination: AnimalCommunity) -> None:
         destination.animal_cohorts[migrant.name].append(migrant)
 
     def migrate_community(self) -> None:
-        """This handles migrating all cohorts in a community."""
+        """This handles migrating all cohorts in a community.
+
+        This migration method initiates migration for two reasons:
+        1) The cohort is starving and needs to move for a chance at resource access
+        2) An initial migration event immediately after birth.
+
+        """
         for cohort in self.all_animal_cohorts:
             if cohort.is_below_mass_threshold(self.constants.dispersal_mass_threshold):
                 # Random walk destination from the neighbouring keys
                 destination_key = choice(self.neighbouring_keys)
                 destination = self.get_destination(destination_key)
                 self.migrate(cohort, destination)
 
+            elif cohort.age == 0.0 and cohort.migrate_juvenile_probability():
+                destination_key = choice(self.neighbouring_keys)
+                destination = self.get_destination(destination_key)
+                self.migrate(cohort, destination)
```

Comment:
There's a lot of code duplication here. Unless you're planning t follow a different migration approach for each branch, I' would just put together all the conditions for migration:
```suggestion
        migrate = (
            cohort.is_below_mass_threshold(self.constants.dispersal_mass_threshold)
            or (cohort.age == 0.0 and cohort.migrate_juvenile_probability())
        )
        
        if not migrate:
            return
            
        destination_key = choice(self.neighbouring_keys)
        destination = self.get_destination(destination_key)
        self.migrate(cohort, destination)
```

Reply:
That is certainly much more straightforward, thanks!